website: add partnerships page (#10384)
hashicorp/nomad: website/content/docs/partnerships.mdx

---
layout: docs
page_title: Nomad Partnerships
description: Update this content
---

# Nomad Integration Program

## Introduction

The HashiCorp Nomad Integration Program allows vendors to integrate their products to work with Nomad. As a scheduler, Nomad encompasses a broad area.  Thereby, a set of integrations may require vendor integration code.
 

> Vendors integrating their solutions via the Nomad Integration Process provide their customers a verified and seamless user experience. The Nomad Integration Program primarily supports coding with the Gco programming language. 

Reviewer: typo?
```suggestion
Vendors integrating their solutions via the Nomad Integration Process provide their customers a verified and seamless user experience. The Nomad Integration Program primarily supports coding with the Go programming language. 
```

5.  Release: Verified integration made available and listed on the HashiCorp website once the HashiCorp technology partnership agreement has been executed
6.  Support: Ongoing maintenance and support of the provider by the vendor.

### 1. Engage
Please begin by providing some basic information about the integration that is being built via a simple [webform](https://docs.google.com/forms/d/e/1FAIpQLSflmhg8lkY5QyH3CB5TUMi4I1ZFQFx_GP_TVDtsbxMaFEMWpw/viewform).

This information is recorded and used by HashiCorp to track the integration through various stages. The information is also used to notify the integration developer of any overlapping work, perhaps coming from the community so you may better focus resources.

Nomad has a large and active community and ecosystem of partners that may have already started working on a similar integration. We'll do our best to connect similar parties to avoid duplicate work.

### 2. Enable

While not mandatory, HashiCorp encourages vendors to sign and MNDA (Mutual Non-Disclosure Agreement) to allow for open dialog and sharing of ideas during the integration process.

> In an effort to support our self-serve model we’ve included links to resources, documentation, examples and best practices to guide you through the Vault integration development and testing process.

Reviewer: making `'` for consistency
```suggestion
In an effort to support our self-serve model we've included links to resources, documentation, examples and best practices to guide you through the Vault integration development and testing process.
```

- Contributing to Nomad [guidelines](https://github.com/hashicorp/nomad/tree/main/contributing)
- [Nomad Developer Community Forum](https://groups.google.com/g/nomad-tool)

> - [Nomad's source code](https://github.com/hashicorp/nomad)
> We encourage vendors to closely follow the above guidance. Adopting the same structure and coding patterns helps expedite the review and release cycles.

Reviewer: ```suggestion
- [Nomad's source code](https://github.com/hashicorp/nomad)

We encourage vendors to closely follow the above guidance. Adopting the same structure and coding patterns helps expedite the review and release cycles.
```

For task, device, storage, and autoscaling plugins, please submit a GitHub pull request (PR) against the [Nomad project](https://github.com/hashicorp/nomad). In some cases the vendor may need to provide HashiCorp with a permanent test account so that the integration can be verified on an ongoing basis.

The review process can take a while to complete and may require some iterations through the code to address and problems identified by the HashiCorp team.

### 5. Release 

At this stage, it is expected that the integration is fully complete, the necessary documentation has been written, the acceptance tests have all passed, and that HashiCorp has reviewed the integration. Once the plugin has been validated and accepted by HashiCorp, the plugin can be hosted anywhere so it can easily be downloaded then installed within Nomad.

Once the integration has been released the vendor is requested to sign the HashiCorp Technology Partner Agreement so that we can have their integration be listed on the HashiCorp website.
 
### 6. Support

> Many vendors view the release step to be the end of the journey, while at HashiCorp we view it to be the beginning of the journey. Getting the integration built is just the first step in enabling users to leverage it against their infrastructure. Once development is completed, on-going effort is required to support the developed integration, maintain the provider and address any issues in a timely manner.

Reviewer: ```suggestion
Many vendors view the release step to be the end of the journey, while at HashiCorp we view it to be the beginning of the journey. Getting the integration built is just the first step in enabling users to leverage it against their infrastructure. Once development is completed, on-going effort is required to support the developed integration, maintain the provider, and address any issues in a timely manner.
```
